src/sentry/api/bases/dashboard.py

```diff
@@ -24,3 +34,23 @@ def _get_dashboard(self, request, organization, dashboard_id):
             id=dashboard_id,
             organization_id=organization.id
         )
+
+
+class OrganizationDashboardWidgetEndpoint(OrganizationDashboardEndpoint):
+    def convert_args(self, request, organization_slug, dashboard_id, widget_id, *args, **kwargs):
+        args, kwargs = super(OrganizationDashboardWidgetEndpoint,
+                             self).convert_args(request, organization_slug, dashboard_id, widget_id, *args, **kwargs)
+
+        try:
+            kwargs['widget'] = self._get_widget(
+                request, kwargs['organization'], dashboard_id, widget_id)
+        except Widget.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return (args, kwargs)
+
+    def _get_widget(self, request, organization, dashboard_id, widget_id):
+        return Widget.objects.get(
+            id=widget_id,
+            dashboard_id=dashboard_id,
```

Comment:
What constrains this to an organization?

Reply:
The dashboard is constrained to the organization. This inherits from `OrganizationDashboardEndpoint` and so, I should probably use the `dashboard` arg created by the parent class rather than `dashboard_id`